### PR TITLE
Use document type to determine the type of the structure

### LINKED ISSUE #8: Articles have page as type in json
When fetching articles the type is incorrect as it returns "page"

### PR DESCRIPTION
**Pull request by wachterjohannes**
Fixes #8

### CODE REVIEW

File: Content/StructureResolver.php [hunk 1]
```diff
@@ -167,10 +178,15 @@ private function getStructureData(StructureInterface $structure): array
...
         }
 
-        $type = 'page';
+        $type = 'unknown';
         if (method_exists($structure, 'getContent') && method_exists($structure->getContent(), 'getTemplateType')) {
             // used for content-bundle to determine current template type
             $type = $structure->getContent()->getTemplateType();
+        } elseif ($document instanceof StructureBehavior) {
+            $type = $this->documentInspector->getMetadata($document)->getAlias();
+            if ('home' === $type) {
```
Comment (wachterjohannes) [inline Content/StructureResolver.php]: @nnatter @alexander-schranz should we differentiate homepage and page inside the Headless-API?

Comment (nnatter) [reply Content/StructureResolver.php]: no i would not use the type `home` anywhere

File: Content/StructureResolver.php [hunk 2]
```diff
@@ -167,10 +178,15 @@ private function getStructureData(StructureInterface $structure): array
             $created = $document->getCreated();
         }
 
-        $type = 'page';
+        $type = 'unknown';
         if (method_exists($structure, 'getContent') && method_exists($structure->getContent(), 'getTemplateType')) {
             // used for content-bundle to determine current template type
```
Comment (nnatter) [inline Content/StructureResolver.php]: can we change this to `determine type for structure that is implemented based on the SuluContentBundle`?

File: Content/StructureResolver.php [hunk 3]
```diff
@@ -167,10 +178,15 @@ private function getStructureData(StructureInterface $structure): array
...
             $type = $structure->getContent()->getTemplateType();
+        } elseif ($document instanceof StructureBehavior) {
+            $type = $this->documentInspector->getMetadata($document)->getAlias();
```
Comment (nnatter) [inline Content/StructureResolver.php]: can we add a comment: `determine type for structure that is implemented in the SuluPageBundle or the SuluArticleBundle`?

File: Tests/Unit/Content/StructureResolverTest.php [hunk 4]
```diff
@@ -48,6 +50,16 @@ class StructureResolverTest extends TestCase
      */
     private $homepageDocument;
 
+    /**
+     * @var Metadata|ObjectProphecy
+     */
+    private $homepageMetadata;
```
Comment (nnatter) [inline Tests/Unit/Content/StructureResolverTest.php]: i think you do not do anything with this variable right now?

File: Tests/Unit/Content/StructureResolverTest.php [hunk 5]
```diff
@@ -48,6 +50,16 @@ class StructureResolverTest extends TestCase
...
+
+    /**
+     * @var Metadata|ObjectProphecy
+     */
+    private $pageMetadata;
```
Comment (nnatter) [inline Tests/Unit/Content/StructureResolverTest.php]: i think you do not do anything with this variable right now?

File: Tests/Unit/Content/StructureResolverTest.php (hunk 4 above)
Comment (wachterjohannes) [reply Tests/Unit/Content/StructureResolverTest.php]: i do :) i do - see the `setUp` method

File: Tests/Unit/Content/StructureResolverTest.php (hunk 5 above)
Comment (wachterjohannes) [reply Tests/Unit/Content/StructureResolverTest.php]: i do :) i do - see the setUp method

File: Content/StructureResolver.php (hunk 1 above)
Comment (wachterjohannes) [reply Content/StructureResolver.php]: 👍

**Comment by wachterjohannes** [comment]
@nnatter fixed

**Pull request merged**